document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/controller/DocumentGeneratorController.java

```diff
@@ -10,53 +11,49 @@
 import org.springframework.web.bind.annotation.ResponseBody;
 import org.springframework.web.bind.annotation.RestController;
 import uk.gov.companieshouse.document.generator.core.models.DocumentRequest;
-import uk.gov.companieshouse.document.generator.core.models.DocumentResponse;
 import uk.gov.companieshouse.document.generator.core.service.DocumentGeneratorService;
+import uk.gov.companieshouse.document.generator.core.service.response.ResponseObject;
+import uk.gov.companieshouse.document.generator.core.utility.ApiResponseMapper;
 import uk.gov.companieshouse.logging.Logger;
 import uk.gov.companieshouse.logging.LoggerFactory;
 
+import javax.servlet.http.HttpServletRequest;
 import javax.validation.Valid;
+import java.util.HashMap;
+import java.util.Map;
 
 import static uk.gov.companieshouse.document.generator.core.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
 
 @RestController
-@RequestMapping(value = "/document-generate", produces = MediaType.APPLICATION_JSON_VALUE)
+@RequestMapping(value = "/private/documents/generate", produces = MediaType.APPLICATION_JSON_VALUE)
 public class DocumentGeneratorController {
 
     private static final Logger LOG = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
 
+    @Autowired
     private DocumentGeneratorService documentGeneratorService;
 
+    @Autowired
+    private ApiResponseMapper apiResponseMapper;
+
     @PostMapping
     @ResponseBody
     public ResponseEntity generateDocument(@Valid @RequestBody DocumentRequest documentRequest,
-                                           BindingResult result) {
+                                           BindingResult result, HttpServletRequest request) {
 
-        DocumentResponse response;
+        String requestId = request.getHeader("X-Request-Id");
 
         if (result.hasErrors()) {
-            LOG.error("error in request body");
+            final Map<String, Object> debugMap = new HashMap<>();
+            debugMap.put("resource_uri", documentRequest.getResourceUrl());
+            debugMap.put("resource_id", documentRequest.getResourceId());
+            LOG.debug("error in request body", debugMap);
```

Comment:
Should this be an error log? If so, `LOG.errorRequest` can be used, taking `request` as a parameter, which adds the requestId to the json blob output in the logs which will help support in debugging.

Reply:
I checked previously with Lead, Log.debug was selected